### ISSUE TITLE
Inconsistent "when" designation between "Keyboard Shortcuts" tab and keybindings.json.

### ISSUE THREAD

**Issue by dbird137** [microsoft/vscode#94633]
Issue Type: <b>Bug</b>

When adding user keybindings through the `Keyboard Shortcuts` settings tab that don't explicitly specify a `when` clause, `"when"` is not added to the `keybindings.json` file and the `when` column in the `Keyboard Shortcuts` tab remains a long dash.  This makes perfect sense.  However, if the desired effect of the keybinding is to always be active, it seems to be insufficient to simply omit the `when` clause in `Keyboard Shortcuts`.  I had to go into `keybindings.json` and add an empty `"when"` string.
```
{
        "key": "ctrl+shift+alt+k",
        "command": "cmake.cleanConfigureAll",
        "when": ""
},
```
After that change, this keybinding seems to work everywhere.  But looking back at `Keyboard Shortcuts` nothing appears to have changed and it still shows a long dash in the `when` column.

This might be a bug with the CMake Tools extension which the command is invoking but regardless, it does seem to demonstrate a VSCode ambiguity in that completely omitting `"when"` in `keybindings.json` like so,
```
{
        "key": "ctrl+shift+alt+k",
        "command": "cmake.cleanConfigureAll"
},
```
does not seem to be equivalent to including it with an empty string in all cases (like in a `terminal` or `output` panel), 
```
{
        "key": "ctrl+shift+alt+k",
        "command": "cmake.cleanConfigureAll",
        "when": ""
},
```

VS Code version: Code 1.43.2 (0ba0ca52957102ca3527cf479571617f0de6ed50, 2020-03-24T07:52:11.516Z)
OS version: Linux x64 5.3.0-46-lowlatency

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz (8 x 3422)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: unavailable_off<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|0, 0, 1|
|Memory (System)|15.52GB (3.67GB free)|
|Process Argv|--no-sandbox --unity-launch|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (35)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-tldr|bmu|1.0.0
better-toml|bun|0.3.2
cmake-format|che|0.6.7
highlight-line-vscode|cli|0.0.1
bracket-pair-colorizer-2|Coe|0.0.29
gitlens|eam|10.2.1
moxer-icons|Equ|5.4.0
vsc-material-theme-icons|equ|1.1.4
file-icons|fil|1.0.23
shell-format|fox|7.0.1
vscode-systemd-support|han|0.1.1
vscode-test-explorer|hbe|2.17.0
output-colorizer|IBM|0.1.2
vscode-rust|kal|0.4.2
cpp-skeleton|lde|0.2.2
hpp-skeleton|lde|0.3.2
python|ms-|2020.3.71659
remote-containers|ms-|0.106.0
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.42.4
vscode-remote-extensionpack|ms-|0.20.0
cmake-tools|ms-|1.3.1
cpptools|ms-|0.27.0
cargo|pan|0.2.3
material-icon-theme|PKi|4.0.1
rust|rus|0.7.0
crates|ser|0.4.9
markdown-preview-enhanced|shd|0.5.2
swdc-vscode|sof|2.1.49
vscode-rust-test-adapter|Swe|0.11.0
tabnine-vscode|Tab|2.6.1
cmake|twx|0.0.17
vscode-icons|vsc|10.0.0
rust-extension-pack|wol|0.1.1

(1 theme extensions excluded)

</details>

**Comment by sandy081** [comment]
>  However, if the desired effect of the keybinding is to always be active, it seems to be insufficient to simply omit the when clause in Keyboard Shortcuts. I had to go into keybindings.json and add an empty "when" string.

@alexdima Please confirm if this is true?

**Comment by dbird137** [comment]
I think I found the problem and it likely has nothing to do with the `"when"` string.

Looking in `Settings>Features>Terminal>Integrated: Commands To Skip Shell`, it looks like the CMake Tools extension adds 4 commands to bypass being sent to the terminal.  However, the command I am using is not one of them.  Adding `cmake.cleanConfigureAll` to the list fixed the problem regardless of whether `"when": ""` was set in keybindings.json.

Since other cmake keybindings that I use were already set (`cmake.build`,`cmake.cleanRebuild`, and `cmake.configure`), all those keybindings did work in an active terminal so I assumed this was a global vscode problem.  I think by editing the keybindings.json, I missed the fact that the terminal was open but no longer "active" which made things work.

Closing this issue.
